main.nf

```diff
@@ -551,7 +551,7 @@ process start_first_batch {
     // Move the first set of sample file into the processing directory
     // so that we jumpstart the workflow.
     sample_files = file("${workflow.workDir}/GEMmaker/stage/*.sample.csv")
-    start_samples = sample_files.sort().take(params.max_cpus)
+    start_samples = sample_files.sort().take(params.max_jobs)
```

Comment:
I don't think we should have a max_jobs argument for GEMmaker.  You are meant to control the size of the parallel jobs using both the `max_cpus` argument that nf-core provides and the QueueSize argument that you can set in a nextflow.config local profile.  See docs here:  https://gemmaker.readthedocs.io/en/latest/execution.html#running-on-a-cluster

Reply:
Palmetto currently has 28,832 CPU cores, so if I set max_cpus to 28,832 and queueSize to 100, GEMmaker will run at most 100 jobs in parallel, but because this line of code will stage up to 28,832 sample files, we will have the same problem we use to have where we have to store intermediate files for thousands of samples even though we're only working on 100 samples at a time.

This line really needs to use queueSize instead of max_cpus.

Reply:
Actually, slight mistake on my part. I should set max_cpus to the max number of cores on a single node on Palmetto, which is 80, So that's not so bad. Even so, I don't think it makes sense for this initial staging process to be tied to the max_cpus param.

Reply:
Sorry I'm thinking in real time here. Actually it is also a problem if I set max_cpus to 80, because then I can't process more than 80 samples in parallel because of how this code works.

Reply:
You just set --max-cpus to the number of parallel jobs you want and set the Nextflow QueueSize to the same number.  The scheduler works out the rest.  That's what we've been doing.

Reply:
I suppose it's not a big deal since none of the gemmaker processes use more than a few cores.